smart-contracts/contracts/mixins/MixinPurchase.sol

```diff
@@ -177,9 +189,17 @@ contract MixinPurchase is
 
     if(minKeyPrice > 0)
     {
-      (unlockDiscount, unlockTokens) = unlockProtocol.computeAvailableDiscountFor(_recipient, minKeyPrice);
-      require(unlockDiscount <= minKeyPrice, 'INVALID_DISCOUNT_FROM_UNLOCK');
-      minKeyPrice -= unlockDiscount;
+      try unlockProtocol.computeAvailableDiscountFor(_recipient, minKeyPrice) 
+      returns (uint unlockDiscount, uint unlockTokens)
+      {
+        require(unlockDiscount <= minKeyPrice, 'INVALID_DISCOUNT_FROM_UNLOCK');
+        minKeyPrice -= unlockDiscount;
+      } catch {
+        // fail silently?
+        // adding an event will require writing to state
+        // also why wasnt this function returning anything?
+        // emit MissingUnlock(address(this), address(unlockProtocol));
```

Comment:
firing an event here will require to remove the `view` modifier from  the function - and likely affects gas. Maybe better to wrap all calls to `recordKeyPurchase` instead?

Reply:
Indeed, I think we would need to wrap more in `recordKeyPurchase`... but also I think for now we can actually remove this `computeAvailableDiscountFor` completely since we don't support it and that locks will be upgradable so we can easily add it back later on (if we need it).
Wdyt?

Reply:
yes, lets removed them